frame/scheduler/src/lib.rs

```diff
@@ -438,6 +438,25 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
+	/// Helper to migrate scheduler when the pallet origin type has changed.
+	pub fn migrate_origin<OldOrigin: Into<T::PalletsOrigin> + codec::Decode>() {
+		Agenda::<T>::translate::<
+			Vec<Option<Scheduled<<T as Trait>::Call, T::BlockNumber, OldOrigin, T::AccountId>>>, _
+		>(|_, agenda| Some(
+			agenda
+				.into_iter()
+				.map(|schedule| schedule.map(|schedule| Scheduled {
+					maybe_id: schedule.maybe_id,
+					priority: schedule.priority,
+					call: schedule.call,
+					maybe_periodic: schedule.maybe_periodic,
+					origin: schedule.origin.into(),
+					_phantom: Default::default(),
+				}))
```

Comment:
```suggestion
				.map(|schedule| schedule.map(|schedule| Scheduled {
					origin: schedule.origin.into(),
					_phantom: Default::default(),
					..scheduled
				}))
```

Reply:
ah actually it doesn't work because it is not the same type 
```rust